src/eduid_userdb/credentials/fido.py

```diff
@@ -34,356 +34,100 @@
-        self._data.update(data)
-
-    @classmethod
-    def from_dict(cls: Type[U2F], data: Dict[str, Any], raise_on_unknown: bool = True) -> U2F:
-        """
-        Construct U2F credential from a data dict.
-        """
-        return cls(data=data, called_directly=False, raise_on_unknown=raise_on_unknown)
+    name_mapping: ClassVar[Dict[str, str]] = {'application': 'created_by'}
 
     @property
-    def key(self):
+    def key(self) -> str:
         """
         Return the element that is used as key.
         """
         return 'sha256:' + sha256(self.keyhandle.encode('utf-8') + self.public_key.encode('utf-8')).hexdigest()
 
-    @property
-    def version(self):
-        """
-        This is the U2F version used by this token.
 
-        :return: U2F version.
-        :rtype: str
-        """
-        return self._data['version']
-
-    @version.setter
-    def version(self, value):
-        """
-        :param value: U2F version. E.g. 'U2F_V2'.
-        :type value: str
-        """
-        if not isinstance(value, string_types):
-            raise UserDBValueError("Invalid 'version': {!r}".format(value))
-        self._data['version'] = value
-
-    @property
-    def attest_cert(self):
-        """
-        The U2F attest_cert from the credential.
-
-        We should probably refine what we store here later on, but for now we just
-        store the whole certificate.
-
-        :return: U2F attest_cert
-        :rtype: str
-        """
-        return self._data['attest_cert']
-
-    @attest_cert.setter
-    def attest_cert(self, value):
-        """
-        :param value: U2F attest_cert.
-        :type value: str
-        """
-        if not isinstance(value, string_types):
-            raise UserDBValueError("Invalid 'attest_cert': {!r}".format(value))
-        self._data['attest_cert'] = value
-
-    @property
-    def public_key(self):
-        """
-        This is the public key of the U2F token.
-
-        :return: U2F public_key.
-        :rtype: str
-        """
-        return self._data['public_key']
-
-    @public_key.setter
-    def public_key(self, value):
-        """
-        :param value: U2F public_key.
-        :type value: str
-        """
-        if not isinstance(value, string_types):
-            raise UserDBValueError("Invalid 'public_key': {!r}".format(value))
-        self._data['public_key'] = value
-
-
-def u2f_from_dict(data, raise_on_unknown=True):
+def u2f_from_dict(data: Dict[str, Any], raise_on_unknown: bool = True) -> U2F:
     """
     Create an U2F instance from a dict.
 
     :param data: Credential parameters from database
     :param raise_on_unknown: Raise UserHasUnknownData if unrecognized data is encountered
-
-    :type data: dict
-    :type raise_on_unknown: bool
-    :rtype: U2F
+                             kept for B/C
     """
-    return U2F.from_dict(data, raise_on_unknown=raise_on_unknown)
+    return U2F.from_dict(data)
 
 
+@dataclass(eq=False, unsafe_hash=True)
```

Comment:
Same question here for eq and unsafe_hash as above.

Maybe credential_data can be used for the `__hash__` method?